src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -26,6 +27,59 @@ public override SarifLogVersionOne VisitSarifLogVersionOne(SarifLogVersionOne no
             return null;
         }
 
+        public FileData TransformFileDataVersionOne(FileDataVersionOne node)
+        {
+            FileData fileData = null;
+
+            if (node != null)
+            {
+                fileData = new FileData
+                {
+                    Length = node.Length,
+                    MimeType = node.MimeType,
+                    Offset = node.Offset,
+                    ParentKey = node.ParentKey,
+                    Properties = node.Properties                    
+                };
+
+                if (fileData.Properties == null)
+                {
+                    fileData.Properties = new Dictionary<string, SerializedPropertyInfo>();
+                }
+
+                if (node.Uri != null)
+                {
+                    fileData.FileLocation = new FileLocation
+                    {
+                        Uri = node.Uri,
+                        UriBaseId = node.UriBaseId
+                    };
+                }
+
+                if (!string.IsNullOrWhiteSpace(node.Contents))
+                {
+                    fileData.Contents = new FileContent
+                    {
+                        Binary = node.Contents
```

Comment:
It's true that a v1 log file doesn't explicitly tell you if the file is text or binary, so "always populate `Binary`" might be a reasonable approach. But you do have the MIME type. So perhaps you could populate `Text` as well for a selection of common textual MIME types. The v1 spec said that `File.Contents` contains the MIME-Base64 encoding of the UTF-8 file contents (if the file is in another encoding, the producer had to transcode it to UTF-8 and then MIME-Base64 encode it).

Reply:
Ah, the heck with it. v1 is just less rich than v2. Let me ping @michaelcfanning to see if he thinks this is worth it.

Reply:
Done, not sure if you'll like the way I did it though...